src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -258,6 +258,9 @@ internal BoundExpression BindToTypeForErrorRecovery(BoundExpression expression,
         /// </summary>
         internal BoundExpression BindToNaturalType(BoundExpression expression, DiagnosticBag diagnostics, bool reportDefaultMissingType = true)
         {
+            if (!expression.NeedsToBeConverted())
```

Comment:
Weren't we asserting that all expressions ended up converted and that `WithWasConverted` was called?

Reply:
No.  In debug builds we do that for locals and parameters.

---
In reply to: [410393631](https://github.com/dotnet/roslyn/pull/40604#discussion_r410393631) [](ancestors = 410393631)